app/common/src/Color.js

```diff
@@ -1,11 +1,17 @@
 // @flow
 
-export default {
+const Color = {
   // brand colors
   // https://images.kiwi.com/content-media/kiwicom_brand_colours.pdf
   brand: '#0097A9',
   brandSecondary: '#0CB3C7',
 
+  icon: {
+    get grey() {
+      return Color.grey.$600;
+    },
+  },
```

Comment:
What is the idea behind? Why do you redefine grey color for the icon? Every grey icon is going to be this grey? This restriction seems a little bit harsh to me. What if I need little bit darker one? Or lighter?

I think it's quite alright just to define the grey color for `TextInput` icon (and even this may be very limiting - see below).

Reply:
I honestly don't think that `icon` should be here because it's not color - should be in the `Icon` component. I removed `TextInput` from this file before - for the same reason. It's not very good design (better to reuse one component and not copy-paste code snippet with a default setting every time). So my idea is that theoretically, you need to use this color only once. But since I have a hard time to convince others, here is an idea how to avoid getters and related boilerplate:

```
const ColorPalette = {
  grey: {
    $100: '#f5f5f5',
    // ...
    $900: '#212121',
  },
};

export default {
  brand: '#0097A9',
  brandSecondary: '#0CB3C7',

  icon: {
    grey: ColorPalette.grey.$600,
  },

  ...ColorPalette,
};
```

cc @VaclavSir 
Related comments: https://github.com/kiwicom/react-native-app/pull/81#discussion_r158594068

Reply:
I would use `Color.grey.$600` directly in the TextInput style, I don't see the benefit of this `Color.icon.grey`. But I agree at least this is better than the getter approach.

Reply:
I think that the idea was really to have unified grey level. Because without `Color.icon.grey` everyone must use exactly `$600` to keep it unified and future changes to all icons are going to be pain (like we need darker grey 🙄 ). So to prevent these stupid mistakes you need to have `Color.grey` or wrap it (or be really careful).